doc/source/whatsnew/v0.21.0.txt

```diff
@@ -109,6 +109,40 @@ For example:
     # the following is now equivalent
     df.drop(columns=['B', 'C'])
 
+.. _whatsnew_0210.enhancements.rename_reindex_axis:
+
+``rename``, ``reindex`` now also accept axis keyword
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :meth:`DataFrame.rename` and :meth:`DataFrame.reindex` methods have gained
+the ``axis`` keyword to specify the axis to target with the operation. This
+harmonizes the method signature of these methods with most of the rest of the
+library (:issue:`12392`).
```

Comment:
I personally wouldn't add the "harmonizes the method signature of these methods with most of the rest of the library " (but that's more personal preference maybe :-) as I said in the issue, IMO it is not comparable with many other functions that use axis as arg, as it is a different use case, it *does* make it consistent with `drop` though, so I would mention that (which also makes it more match with the whatsnew entry about drop above))